lib/cc/analyzer/issue_validator.rb

```diff
@@ -0,0 +1,49 @@
+module CC
+  module Analyzer
+    class IssueValidator
+      CHECKS = [
+        IssueCategoryValidation,
+        IssueCheckNamePresenceValidation,
+        IssueDescriptionPresenceValidation,
+        IssuePathExistenceValidation,
+        IssuePathPresenceValidation,
+        IssueRelativePathValidation,
+        IssueTypeValidation,
+      ].freeze
+
+      attr_reader :error
+
+      def initialize(issue)
+        @issue = issue
+      end
+
+      def valid?
+        @error.blank?
```

Comment:
API nit: most `ActiveModel`-esque interfaces allow you to call `#valid?` without explicitly calling `validate`. Should we do the same? It's our own little world, so we don't have to, but OTOH it seems like something likely to trip somebody up down the road, and IMHO it is a nice little feature.